src/predictors/historic-average-predictor.cc

```diff
@@ -0,0 +1,139 @@
+#include "predictors/historic-average-predictor.h"
+#include "predictors/predictor-utils.h"
+
+#include <iostream>
+#include <iomanip>
+#include <ctime>
+#include <sstream>
+
+#include <boost/geometry/algorithms/distance.hpp>
+#include <boost/geometry/geometries/point.hpp>
+#include <boost/geometry/strategies/spherical/distance_haversine.hpp>
+
+HistoricAveragePredictor::HistoricAveragePredictor() = default;
+
+/**
+ * Predictor based on the HistoricAveragePredictor approach
+ * This checks when the trip reached the following trip in the last n trips
+ * and uses this as a prediction.
+ *
+ * @param tripUpdates current tripUpdate feed to be updated
+ * @param vehiclePositions positions of vehicles as feed
+ * @param timetable timetable to match the vehiclePositions to stops
+ */
+void HistoricAveragePredictor::predict(
+    transit_realtime::FeedMessage& tripUpdates,
+    const transit_realtime::FeedMessage& vehiclePositions,
+    const nigiri::timetable& timetable) {
+  // Part 1: Storing of departures based on GTFS-Position-Tracker
+  auto now = std::chrono::system_clock::now();
+  auto current_time = std::chrono::duration_cast<std::chrono::seconds>(now.time_since_epoch()).count();
+  auto today = std::format("{:%Y-%m-%d}", now);
+
+  for (const transit_realtime::FeedEntity& entity : vehiclePositions.entity()) {
+    // For this prototype we only care about vehicle positions. Service alerts and other trip updates are ignored
+    if (entity.has_vehicle()) {
+      const transit_realtime::VehiclePosition& vehicle_position = entity.vehicle();
+      // Get Trip ID
+      std::string tripID = vehicle_position.trip().trip_id();
+      std::string routeID = vehicle_position.trip().route_id();
+      std::string vehicleID = vehicle_position.vehicle().id();
+
+      // Get a stop list for a given trip
+      std::vector<nigiri::location> stop_list = predictorUtils::get_stops_for_trip(timetable, tripID);
+      // check for each stop if we are close
+      for (nigiri::location location : stop_list) {
```

Comment:
```suggestion
      std::vector<nigiri::location> stopList = predictorUtils::get_stops_for_trip(timetable, tripID);
      // check for each stop if we are close
      for (nigiri::location location : stopList) {
```

Of course check that it is everywhere correct